src/main/java/io/github/thebusybiscuit/slimefun4/core/guide/options/SlimefunGuideSettings.java

```diff
@@ -232,21 +233,48 @@ private static void addConfigurableOptions(Player p, ChestMenu menu, ItemStack g
      * in their {@link SlimefunGuide}.
      * If they enabled this setting, they will see fireworks when they unlock a {@link Research}.
      * 
-     * @param p
-     *            The {@link Player}
+     * @param p The {@link Player}
      * 
      * @return Whether this {@link Player} wants to see fireworks when unlocking a {@link Research}
      */
     public static boolean hasFireworksEnabled(@Nonnull Player p) {
+        return getOptionValue(p, FireworksOption.class, true);
+    }
+
+    /**
+     * This method checks if the given {@link Player} has enabled the {@link LearningAnimationOption}
+     * in their {@link SlimefunGuide}.
+     * If they enabled this setting, they will see messages in chat about the progress of their {@link Research}.
+     *
+     * @param p The {@link Player}
+     *
+     * @return Whether this {@link Player} wants to info messages in chat when unlocking a {@link Research}
+     */
+    public static boolean hasLearningAnimationEnabled(@Nonnull Player p) {
+        return getOptionValue(p, LearningAnimationOption.class, true);
+    }
+
+    /**
+     * Helper method to get the value of a {@link SlimefunGuideOption} that the {@link Player}
+     * has set in their {@link SlimefunGuide}
+     *
+     * @param p The {@link Player}
+     * @param cls Class of the {@link SlimefunGuideOption} to get the value of
+     * @param defaultValue Default value to return in case the option is not found at all or has no value set
+     * @param <T> Type of the {@link SlimefunGuideOption} value
+     * @param <OPTION> Type of the {@link SlimefunGuideOption}
+     * @return The value of given {@link SlimefunGuideOption}
+     */
+    private static <T, OPTION extends SlimefunGuideOption<T>> T getOptionValue(@Nonnull Player p, @Nonnull Class<OPTION> cls, T defaultValue) {
         for (SlimefunGuideOption<?> option : options) {
-            if (option instanceof FireworksOption) {
-                FireworksOption fireworks = (FireworksOption) option;
+            if (cls.isInstance(option)) {
+                OPTION o = (OPTION) option;
                 ItemStack guide = SlimefunGuide.getItem(SlimefunGuideMode.SURVIVAL_MODE);
-                return fireworks.getSelectedOption(p, guide).orElse(true);
+                return o.getSelectedOption(p, guide).orElse(defaultValue);
             }
         }
 
-        return true;
+        return defaultValue;
     }
```

Comment:
The use of T and OPTION here as generic types is kinda unsettling to me.

I would rather suggest to use T and V or K and V (for type/key and value) and putting the options type first.
This seems confusing right now.

There should also be an annotation on the method to indicate whether this is nullable or not.